### PR TITLE
Emissive Cameras

### PR DESCRIPTION
**Pull request by Xander3359**
## About The Pull Request
image
The eye of the camera is glowy now, very cool.
## Why It's Good For The Game
I think it's neat
## Changelog
:cl:
imageadd: Adds camera emissives
/:cl:

### CODE REVIEW

File: code/game/objects/machinery/camera/camera.dm [hunk 1]
```diff
@@ -310,10 +309,21 @@
...
-
 /obj/machinery/camera/autoname
+	light_range = 1
+	light_power = 0.2
 	var/number = 0 //camera number in area
 
+/obj/machinery/camera/autoname/wirecutter_act(mob/living/user, obj/item/I)
+	. = ..()
+	if(.)
+		set_light(initial(light_range), initial(light_power))
```
Comment (Lumipharon) [inline code/game/objects/machinery/camera/camera.dm:320]: Why is this just turning the light to initial? snipping can turn it on or off.

**Review by Lumipharon — CHANGES_REQUESTED** [review]

File: code/game/objects/machinery/camera/camera.dm [hunk 2]
```diff
@@ -234,7 +233,7 @@
 		else
 			myarea = null
 	else
-		set_light(0)
+		set_light(initial(light_range), initial(light_power))
```
Comment (Lumipharon) [inline code/game/objects/machinery/camera/camera.dm:236]: this is turning the camera off. Why are you changing this to inital in this case?

Comment (Xander3359) [reply code/game/objects/machinery/camera/camera.dm:236]: This is for when the AI turns off their lights. It should be setting to initial so the emissives work as discussed (minimum light requirements)

File: code/game/objects/machinery/camera/camera.dm [hunk 3]
```diff
@@ -296,7 +295,7 @@
 	if(on)
 		set_light(AI_CAMERA_LUMINOSITY, AI_CAMERA_LUMINOSITY)
```
Comment (ZeWaka) [inline code/game/objects/machinery/camera/camera.dm:296]: wat, it's `set_light(l_range, l_power,...)` this doesn't make sense

Comment (Xander3359) [reply code/game/objects/machinery/camera/camera.dm:296]: Both are 6

Comment (ZeWaka) [reply code/game/objects/machinery/camera/camera.dm:296]: and? still dumb AF, the defines are being abused

File: code/game/objects/machinery/camera/camera.dm (hunk 1 above)
Comment (Lumipharon) [reply code/game/objects/machinery/camera/camera.dm]: This is not fixed

File: code/game/objects/machinery/camera/camera.dm [hunk 4]
```diff
@@ -310,10 +309,21 @@
 	user.see_in_dark = 2
 	return TRUE
 
-
 /obj/machinery/camera/autoname
+	light_range = 1
+	light_power = 0.2
```
Comment (psykzz) [inline code/game/objects/machinery/camera/camera.dm:314]: Does this need adding?

Comment (Xander3359) [reply code/game/objects/machinery/camera/camera.dm:314]: Yes, emissives won't glow in the dark if there isn't a minimum amount of light level.
When light is at 0, humans can't see it (no night vision), which defeats the purpose.
I wanted the green bits of the camera eye to be visible in darkness.

**Review by Lumipharon — APPROVED** [review]

**Pull request merged**